site/src/contexts/ProxyContext.tsx

```diff
@@ -45,15 +80,26 @@ export const ProxyContext = createContext<ProxyContextValue | undefined>(
  */
 export const ProxyProvider: FC<PropsWithChildren> = ({ children }) => {
   // Try to load the preferred proxy from local storage.
-  let savedProxy = loadPreferredProxy()
+  const [savedProxy, setUserProxy] = useState<Region | undefined>(
```

Comment:
Why do we need to `useState` here? Isn't this state already saved in local storage? [I see you're allowing the context consumer](https://github.com/coder/coder/pull/7515/files#diff-1beec2c2724792e5a29bc182e07983cd672b52ea138eca517ddb65c5e30bb500R180) to `setUserProxy`, but in doing so, we're really updating state in two places (local storage and this `setState`. I think it would be better to let local storage be the main storage mechanism here.

Reply:
Local storage is my persistent mechanism for storing and reloading state (state being the user selection).

This is the context mechanism for storing and reloading state all in memory.

The reason I use both is because I wanted the selected proxy to be a dependency for the `useEffect`, and didn't know how to make the value stored in local storage trigger a new `useEffect` render.